asl-translator.cabal

```diff
@@ -39,7 +39,7 @@ common shared-properties
                        ilist,
                        integer-logarithms,
                        bimap,
-                       text >= 1 && < 2,
+                       text >= 1 && < 2.1,
```

Comment:
I know this is not the first PR to make this change, but seeing it repeatedly I'm now wondering: since we constrain `base` (not tightly, but we do), and since `text` is a boot library (i.e. builtin to the GHC distribution, per https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/libraries/version-history), should we just remove constraints on `text` altogether?

Reply:
GHC does ship with `text`, but strictly speaking, it's not tied directly to the GHC version. If you want, you can choose a build plan that picks a different version of `text` than what GHC ships with. (I had to do quite a bit of testing of this sort with `text-2.0` in my own libraries before it was released.) For this reason, I think it's a good idea to be cautious with upper version bounds, although I don't feel too strongly about this.

Reply:
OK.  I'm fine with keeping the constraints... I was just thinking about ways to minimize our churn going forward.